orderbook/src/storage.rs

```diff
@@ -0,0 +1,34 @@
+mod memory;
+mod postgresql;
+
+use anyhow::Result;
+use contracts::GPv2Settlement;
+use model::order::{Order, OrderCreation, OrderUid};
+
+pub use memory::OrderBook as InMemoryOrderBook;
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum AddOrderResult {
+    Added(OrderUid),
+    DuplicatedOrder,
+    InvalidSignature,
```

Comment:
It works out better to have `Added` be part of this enum instead of being the Ok side of the `add_order`. This allows makes the error for `add_order` always an internal error (like not being able to reach the database).